server/message_queue_service.py

```diff
@@ -11,6 +11,10 @@
 from .decorators import with_logger
 
 
+class ConnectionAttemptFailed(Exception):
```

Comment:
Maybe inherit from ConnectionError?

```suggestion
class ConnectionAttemptFailed(ConnectionError):
```